mat2vec/training/phrase2vec.py

```diff
@@ -57,64 +73,108 @@ def wordgrams(sent, depth, pc, th, ct, et, ip, d=0):
     parser.add_argument("--alpha", default=0.01, help="Learning rate.")
     parser.add_argument("--batch", default=10000, help="Minibatch size.")
-    parser.add_argument("--negative", default=15, help="Number of negative samples.")
-    parser.add_argument("--subsample", default=0.0001, help="Subsampling rate.")
-    parser.add_argument("--phrase_depth", default=2, help="The number of passes to perform for phrase generation.")
-    parser.add_argument("--phrase_count", default=10, help="Minimum number of occurrences for phrase to be considered.")
-    parser.add_argument("--phrase_threshold", default=15.0, help="Phrase importance threshold.")
-    parser.add_argument("-include_extra_phrases",
-                        action="store_true",
-                        help="If true, will look for all_ents.p and add extra phrases.")
-    parser.add_argument("-sg", action="store_true", help="If set, will train a skip-gram, otherwise a CBOW.")
-    parser.add_argument("-hs", action="store_true", help="If set, hierarchical softmax will be used.")
-    parser.add_argument("-keep_formula", action="store_true",
-                        help="If set, keeps simple chemical formula independent on count.")
-    parser.add_argument("-notmp", action="store_true", help="If set, will not store the progress in tmp folder.")
+    parser.add_argument(
+        "--negative",
+        default=15,
+        help="Number of negative samples.")
+    parser.add_argument(
+        "--subsample",
+        default=0.0001,
+        help="Subsampling rate.")
+    parser.add_argument(
+        "--phrase_depth",
+        default=2,
+        help="The number of passes to perform for phrase generation.")
+    parser.add_argument(
+        "--phrase_count",
+        default=10,
+        help="Minimum number of occurrences for phrase to be considered.")
+    parser.add_argument(
+        "--phrase_threshold",
+        default=15.0,
+        help="Phrase importance threshold.")
+    parser.add_argument(
+        "-include_extra_phrases",
+        action="store_true",
+        help="If true, will look for all_ents.p and add extra phrases.")
+    parser.add_argument(
+        "-sg",
+        action="store_true",
+        help="If set, will train a skip-gram, otherwise a CBOW.")
+    parser.add_argument(
+        "-hs",
+        action="store_true",
+        help="If set, hierarchical softmax will be used.")
+    parser.add_argument(
+        "-keep_formula",
+        action="store_true",
+        help="If set, keeps simple chemical formula independent on count.")
+    parser.add_argument(
+        "-notmp",
+        action="store_true",
+        help="If set, will not store the progress in tmp folder.")
     args = parser.parse_args()
 
     all_formula = []
     if args.keep_formula:
         try:
-            all_formula = load_obj(args.corpus + "_formula")  # list of formula is supplied
+            # list of formula is supplied
```

Comment:
Some of these code should probably be grouped and placed somewhere else.

Reply:
Yeah, it is a mess. Thanks for suggesting this!